kv/kvserver: TestLeasePreferencesDuringOutage failed

kv/kvserver.TestLeasePreferencesDuringOutage [failed](https://tanzanite.cluster.engflow.com/invocations/default/ececd385-b0fb-41a2-9179-a06488279574?testReportRun=3&testReportShard=26&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-25.1 @ [5af422557eaec1c2d34fdc9524fe25611bef1c01](https://github.com/cockroachdb/cockroach/commits/5af422557eaec1c2d34fdc9524fe25611bef1c01):


```
=== RUN   TestLeasePreferencesDuringOutage
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeasePreferencesDuringOutage909648597
    test_log_scope.go:76: use -show-logs to present logs inline
    client_lease_test.go:1179: 
        	Error Trace:	pkg/kv/kvserver/client_lease_test.go:1179
        	            				pkg/kv/kvserver/client_lease_test.go:1183
        	            				pkg/kv/kvserver/client_lease_test.go:1216
        	Error:      	Received unexpected error:
        	            	heartbeat failed on epoch increment
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness.init
        	            	  | 	pkg/kv/kvserver/liveness/liveness.go:98
        	            	  | runtime.doInit1
        	            	  | 	GOROOT/src/runtime/proc.go:7206
        	            	  | runtime.doInit
        	            	  | 	GOROOT/src/runtime/proc.go:7173
        	            	  | runtime.main
        	            	  | 	GOROOT/src/runtime/proc.go:253
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) heartbeat failed on epoch increment
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestLeasePreferencesDuringOutage
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeasePreferencesDuringOutage909648597
--- FAIL: TestLeasePreferencesDuringOutage (5.29s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=26</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeasePreferencesDuringOutage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48515

--- a/pkg/kv/kvserver/client_lease_test.go
+++ b/pkg/kv/kvserver/client_lease_test.go
@@ -1178,7 +1178,17 @@ func TestLeasePreferencesDuringOutage(t *testing.T) {
 			// allocator on server 0 may see everyone as temporarily dead due to the
 			// clock move above.
 			for _, i := range []int{0, 3, 4} {
-				require.NoError(t, tc.Servers[i].HeartbeatNodeLiveness())
+				testutils.SucceedsSoon(t, func() error {
+					err := tc.Servers[i].HeartbeatNodeLiveness()
+					if err != nil {
+						if errors.Is(err, liveness.ErrEpochIncremented) {
+							t.Logf("retrying heartbeat after err %s", err)
+							return err
+						}
+						t.Fatalf("unexpected error heartbeating liveness record for server %d: %s", i, err)
+					}
+					return nil
+				})
 				require.NoError(t, tc.GetFirstStoreFromServer(t, i).GossipStore(ctx, true))
 			}
 		}

